registration-processor/registration-processor-registration-status-service-impl/src/main/java/io/mosip/registration/processor/status/service/impl/SyncRegistrationServiceImpl.java

```diff
@@ -1,745 +1,821 @@
+		registrationStatusDto.setAdditionalInfoReqId(entity.getAdditionalInfoReqId());
+		registrationStatusDto.setStatusCode(RegistrationExternalStatusCode.UPLOAD_PENDING.toString());
+		return registrationStatusDto;
+	}
+
+	@Override
+	public boolean deleteAdditionalInfo(SyncRegistrationEntity syncEntity) {
+		return syncRegistrationDao.deleteAdditionalInfo(syncEntity);
+	}
+
+	@Override
+	public List<PacketStatusDTO> getByPacketIdsWithStatus(List<PacketStatusSubRequestDTO> packetIdsRequested) {
+		regProcLogger.debug(LoggerFileConstant.SESSIONID.toString(), LoggerFileConstant.USERID.toString(), "",
+				"SyncRegistrationServiceImpl::getByPacketIdsWithStatus()::entry");
+
+		try {
+			List<String> packetIds = new ArrayList<>();
+
+			for (PacketStatusSubRequestDTO packetStatusSubRequestDTO : packetIdsRequested) {
+				packetIds.add(packetStatusSubRequestDTO.getPacketId());
+			}
+			if (!packetIds.isEmpty()) {
+				List<SyncRegistrationEntity> syncRegistrationEntityList = syncRegistrationDao
+						.getByPacketIds(packetIds);
+
+				regProcLogger.debug(LoggerFileConstant.SESSIONID.toString(), LoggerFileConstant.USERID.toString(), "",
+						"SyncRegistrationServiceImpl::getByPacketIdsWithStatus()::exit");
+				return convertEntityListToDtoListAndGetExternalStatusForPacket(syncRegistrationEntityList);
+			}
+			return null;
+		} catch (DataAccessLayerException e) {
+
+			regProcLogger.error(LoggerFileConstant.SESSIONID.toString(), LoggerFileConstant.REGISTRATIONID.toString(),
+					"", e.getMessage() + ExceptionUtils.getStackTrace(e));
+			throw new TablenotAccessibleException(
+					PlatformErrorMessages.RPR_RGS_REGISTRATION_TABLE_NOT_ACCESSIBLE.getMessage(), e);
+		}
+	}
+
+	private List<PacketStatusDTO> convertEntityListToDtoListAndGetExternalStatusForPacket(
+			List<SyncRegistrationEntity> syncRegistrationEntityList) {
+		List<PacketStatusDTO> list = new ArrayList<>();
+		if (syncRegistrationEntityList != null) {
+			for (SyncRegistrationEntity entity : syncRegistrationEntityList) {
+				list.add(convertEntityToDtoAndGetExternalStatusForPacket(entity));
+			}
+
+		}
+		return list;
+	}
+
+	private PacketStatusDTO convertEntityToDtoAndGetExternalStatusForPacket(SyncRegistrationEntity entity) {
+		PacketStatusDTO packetStatusDTO = new PacketStatusDTO();
+		packetStatusDTO.setPacketId(entity.getPacketId());
+		packetStatusDTO.setStatusCode(RegistrationExternalStatusCode.UPLOAD_PENDING.toString());
+		return packetStatusDTO;
+	}
+
+	@Override
+	public List<SyncRegistrationEntity> getByPacketIds(List<PacketStatusSubRequestDTO> packetIdsRequested) {
+		regProcLogger.debug(LoggerFileConstant.SESSIONID.toString(), LoggerFileConstant.USERID.toString(), "",
+				"SyncRegistrationServiceImpl::getByPacketIds()::entry");
+
+		try {
+			List<String> packetIds = new ArrayList<>();
+
+			for (PacketStatusSubRequestDTO packetStatusSubRequestDTO : packetIdsRequested) {
+				packetIds.add(packetStatusSubRequestDTO.getPacketId());
+			}

```

Comment:
Lets not pass a request dto to a service, can we rather exact the packet id in the controller and receiver the list of string packet ids and pass to this method ?

Reply:
fixed